GSA/digitalgov.gov
# Add prettier for Hugo templates

## Summary

This PR does the following:
- Adds NPM scripts for testing & fixing template formatting
- Adds formatting check to CircleCI
- Fixes errors found
- Simplified logic in topics list at bottom of `/topics`

This is not a breaking change. Created documentation in google drive for prettier commands and common types of errors.

### Preview

- [Topics list](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/jm-prettier-hugo/topics/) [4877bf3]
- [Accordion updates](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/jm-prettier-hugo/event/2022/09/15/uswds-monthly-call-september-2022/). The icon passed to shortcode is set in a variable first, then passed through relUrl filter to avoid encoding issues. There should be **no** regressions to accordion shortcodes. [a9bccc3]
- [Accordion example 2](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/jm-prettier-hugo/style-guide/#:~:text=as%20%E2%80%9CDigital.gov%E2%80%9D.-,Logos,-Expand%20for%20digital)
- [Do/Don't table](https://federalist-466b7d92-5da1-4208-974f-d61fd4348571.sites.pages.cloud.gov/preview/gsa/digitalgov.gov/jm-prettier-hugo/resources/do-dont-table/)


### Solution

1. Added prettier w/ Hugo plugin
2. Ran prettier write to fix templates - `npx prettier -w ./themes/**/*.html`
3. Ran prettier check to view errors - `npx prettier -c ./themes/**/*.html`
4. Fixed HTML errors (extra tags) ex: [d9429f9]


### How To Test

1. NPM script `test:templates` should check Hugo templates, without errors.
5. CircleCI will check for formatting on builds  in new `Template formatting` task

    <details><summary>Build 34593</summary>

      [Build 34593 link →](https://app.circleci.com/pipelines/github/GSA/digitalgov.gov/13881/workflows/87ad71dd-e038-4ee2-ba18-83241770d608/jobs/34594)

      ![image](https://user-images.githubusercontent.com/3385219/231773449-98539a71-14e1-4e0d-b72b-157859f166b6.png)
    </details> 

#### Topics refactor

1. There should be no visual regressions in `/topics` page
2. Topics should be one column on mobile & three on desktop

## New NPM scripts

**prettier:templates**
Used to run prettier on templates.

**test:templates**
Runs prettier with check flag to test formatting locally and in CircleCI.

## Dependencies

| Dependency | Version | Description |
|--------|--------|--------|
| prettier | 2.8.7 | Used for formatting. |
| prettier-plugin-go-template | 0.0.13 | Plugin for formatting Hugo templates. |

---

### Dev Checklist
